sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -962,8 +962,8 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
 
   test("SET commands semantics using sql()") {
     spark.sessionState.conf.clear()
-    val testKey = "test.key.0"
-    val testVal = "test.val.0"
+    val testKey = "test.key.k0"
```

Comment:
This is a good catch. Shall we support it to maximize the backward compatibility?
```
configIdentifier
    : IDENTIFIER
    | nonReserved
    | strictNonReserved
    | DIGIT+
    ;
```

Reply:
sure.

Reply:
I'm still looking into this.

Reply:
I tried it though, I found just adding `DIGIT+`([Actually, I added `INTEGER_VALUE` instead](https://github.com/apache/spark/pull/29146/files#diff-8c1cb2af4aa1109e08481dae79052cc3R280)) there did not work well...
For example, a pattern below was okay;
```
SET spark:sql:3=value or SET 1:2:key=value
```
https://github.com/apache/spark/pull/29146/files#diff-e9f35fff083788e6494cb1220a792b99R90-R92

But, a following pattern couldn't be parsed;
```
SET spark.sql.3=value or SET 1.2.key=value
```
https://github.com/apache/spark/pull/29146/files#diff-e9f35fff083788e6494cb1220a792b99R105-R106

That's because `.3` and `1.2` are regarded as  the `DECIMAL_VALUE` tokens, so the `configKey` parsing rule cannot match them correctly. I looked for an approach to avoid it, but I haven't found a good idea so far.

Any idea for this? Anyway, I will keep looking for it.

Reply:
damn this is really hard. We probably should simplify the parser rule
```
SET key=quotedIdentifier (EQ vaule=.*)?  #simpleSetConfig
SET mixed=.*              #mixedSetConfig
```

In the mixed case, we split the string by `=`, and make sure the first part(key) has no special chars.

Reply:
hm, okay. I'll look into this more.